src/main/java/org/terasology/gooeyDefence/PathfindingSystem.java

```diff
@@ -75,20 +81,48 @@ public void onChangedBlock(OnChangedBlock event, EntityRef entity) {
         }
     }
 
+    /**
+     * Called to request an enemy be re-pathed.
+     *
+     * @see RepathEnemyRequest
+     */
+    @ReceiveEvent
+    public void onRepathEnemyRequest(RepathEnemyRequest event, EntityRef entity, LocationComponent locationComponent) {
+        /* Pause the enemy */
+        entity.removeComponent(EnemyManager.getPathComponent(entity).getClass());
+        entity.addComponent(new BlankPathComponent(new Vector3i(locationComponent.getWorldPosition())));
+
+        /* Process its path */
+        calculatePath(new Vector3i(locationComponent.getWorldPosition()), (path, end) -> {
+            if (path.size() != 0) {
+                CustomPathComponent customPathComponent = new CustomPathComponent(path);
+                entity.addComponent(customPathComponent);
+                entity.removeComponent(BlankPathComponent.class);
+            }
+        });
+    }
+
     /**
      * Calculate the path from an entrance to the centre
      *
      * @param id       The entrance to calculate from
-     * @param callback A callback to be invoked after all pending path calculations have completed
+     * @param callback A callback to be invoked after the path calculation has finished.
      */
     private void calculatePath(int id, Runnable callback) {
```

Comment:
so i guess paths can be referred to by both a pathId and entranceId (which are the same values)? I think paths can be referred to by their entranceId everywhere. so in this case id would be renamed to entranceId and the javadoc would become redundant

Reply:
This will be fixed in PR #5 to avoid merge conflicts